extensions/amp-image-viewer/0.1/amp-image-viewer.js

```diff
@@ -261,104 +245,121 @@ export class AmpImageViewer extends AMP.BaseElement {
-            ./*OK*/getBoundingClientRect());
-
-        const sourceAspectRatio = this.sourceWidth_ / this.sourceHeight_;
-        let height = Math.min(this.elementBox_.width / sourceAspectRatio,
-            this.elementBox_.height);
-        let width = Math.min(this.elementBox_.height * sourceAspectRatio,
-            this.elementBox_.width);
-
-        if (Math.abs(width - this.sourceWidth_) <= 16
-          && Math.abs(height - this.sourceHeight_ <= 16)) {
-          width = this.sourceWidth_;
-          height = this.sourceHeight_;
-        }
+  measure_() {
+    this.elementBox_ = layoutRectFromDomRect(this.element
+        ./*OK*/getBoundingClientRect());
+
+    const sourceAspectRatio = this.sourceWidth_ / this.sourceHeight_;
+    let height = Math.min(this.elementBox_.width / sourceAspectRatio,
+        this.elementBox_.height);
+    let width = Math.min(this.elementBox_.height * sourceAspectRatio,
+        this.elementBox_.width);
+
+    if (Math.abs(width - this.sourceWidth_) <= 16
+    && Math.abs(height - this.sourceHeight_ <= 16)) {
+      width = this.sourceWidth_;
+      height = this.sourceHeight_;
+    }
 
-        this.imageBox_ = layoutRectLtwh(
-            Math.round((this.elementBox_.width - width) / 2),
-            Math.round((this.elementBox_.height - height) / 2),
-            Math.round(width),
-            Math.round(height));
-
-        // Adjust max scale to at least fit the screen.
-        const elementBoxRatio = this.elementBox_.width
-          / this.elementBox_.height;
-        const maxScale = Math.max(
-            elementBoxRatio / sourceAspectRatio,
-            sourceAspectRatio / elementBoxRatio
-        );
-        this.maxScale_ = Math.max(DEFAULT_MAX_SCALE, maxScale);
-
-        // Reset zoom and pan.
-        this.startScale_ = this.scale_ = 1;
-        this.startX_ = this.posX_ = 0;
-        this.startY_ = this.posY_ = 0;
-        this.updatePanZoomBounds_(this.scale_);
-      },
-      mutate: () => {
+    this.imageBox_ = layoutRectLtwh(
+        Math.round((this.elementBox_.width - width) / 2),
+        Math.round((this.elementBox_.height - height) / 2),
+        Math.round(width),
+        Math.round(height));
+
+    // Adjust max scale to at least fit the screen.
+    const elementBoxRatio = this.elementBox_.width
+    / this.elementBox_.height;
+    const maxScale = Math.max(
+        elementBoxRatio / sourceAspectRatio,
+        sourceAspectRatio / elementBoxRatio
+    );
+    this.maxScale_ = Math.max(DEFAULT_MAX_SCALE, maxScale);
+
+    // Reset zoom and pan.
+    this.startScale_ = this.scale_ = 1;
+    this.startX_ = this.posX_ = 0;
+    this.startY_ = this.posY_ = 0;
+    this.updatePanZoomBounds_(this.scale_);
+  }
+
+  /**
+   * Measures and resets the image dimensions, after the element
+   * dimensions changes.
+   * @return {!Promise}
+   */
+  resetImageDimensions_() {
+    return this.measureElement(() => this.measure_()).then(() => {
```

Comment:
This should probably be `measureMutateElement` since you are doing both here.

Reply:
This is a bit weird, but what I'm doing here is measuring the entire element, but only mutating `this.image_` which is a child of the element. So technically I don't know if they can go in the same `measureMutateElement` block because they apply to different elements.

Reply:
Okay, conferred with @jridgewell , and the TLDR is that `onLayoutMeasure` is not intended for triggering resize logic. Added a comment to refactor resize logic out of `onLayoutMeasure` into `onMeasureChange` when that is ready. This right now is a hack to avoid triggering an infinite loop that breaks `<amp-lightbox-gallery>`.